base/server/cmscore/src/com/netscape/cmscore/dbs/DBVirtualList.java

```diff
@@ -437,6 +438,11 @@ private synchronized boolean getEntries() {
             String ldapAttrs[] = null;
             LDAPSearchResults result;
 
+            Integer size_limit = (Integer) mConn.getOption(LDAPv2.SIZELIMIT);
+            if (mPageSize < size_limit) {
```

Comment:
Do we need to worry about 0 or negative mPageSize?

Reply:
Uh....... That's a very interesting question. `mPageSize` **can** be negative, depending on the constructor (and `setPageSize(...)` doesn't check negativity and seems overly complex). 

On the other hand, the actual API takes `maxCount` and also doesn't check negatives. In fact, it actually sets `mPageSize` to be negative in the constructor in some places (when doing reverse listings).

I think this ultimately needs a `Math.abs(mPageSize)`, and yes, different handling when `size_limit == 0`.